case_studies/strong_lensing/run_sim.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+# Script to automate running the galsim command with the yaml configuration
+
+# Default Values
+num_files=3
+num_galaxies=3
+img_size=500
+
+
+while getopts "f:s:g:" opt; do
+  case $opt in
+    f) num_files="$OPTARG"  # num output images
+    ;;
+    s) img_size="$OPTARG" # size of output images
+    ;;
+    g) num_galaxies="$OPTARG" # number of galaxies in final output image
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+    exit 1
+    ;;
+  esac
+
+  case $OPTARG in
+    -*) echo "Option $opt needs a valid argument"
+    exit 1
+    ;;
+  esac
+done
+
+
+# base image config file
+GALAXY_CONFIG_FILE="galsim-random.yaml"
+# final image config file
+IMAGE_CONFIG_FILE="simulate.yaml"
+
+
+NUM_ITERATIONS=$((num_galaxies+1))
+
+# Set the output directory for base images
+OUTPUT_DIR="data/images"
+
+# Output directory for final images
+FINAL_OUTPUT_DIR="output_yaml"
+
+# Catalog Directory
+CATALOG_DIR="catalogs"
+if [ ! -d "$OUTPUT_DIR" ]; then
+  mkdir -p "$CATALOG_DIR"
+fi
+
+MAG=0.029999
+# Create the output directory if it doesn't exist
+if [ ! -d "$OUTPUT_DIR" ]; then
+    mkdir -p "$OUTPUT_DIR"
+    if [ $? -ne 0 ]; then
+        echo "Failed to create output directory $OUTPUT_DIR!"
+        exit 1
+    fi
+fi
+
+# Log file
+LOG_FILE="galsim_run.log"
+
+NUM_UNLENSED_IMAGES=$((num_galaxies/2))
+NUM_LENSED_IMAGES=$((num_galaxies-NUM_UNLENSED_IMAGES))
+
+echo "Starting data generation..."
+echo "Data generation beginning..." &> $LOG_FILE
```

Comment:
Would you display error messages in addition to directing them to a log file?

Reply:
Actually, how about just displaying all the output to STDOUT? Otherwise the program appears to be hanging

Reply:
(A user can always redirect STDOUT to a file if they want, and then they'll know where to find the output.)